packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py

```diff
@@ -79,3 +91,140 @@ async def test_check_for_aws_http_errors(
+        aws_access_key_id="xxx",
+    ) as client:
+        yield client
+
+
+async def _clean_bucket_content(s3_client: AioBaseClient, bucket: str):
+    response = await s3_client.list_objects_v2(Bucket=bucket)
+    while response["KeyCount"] > 0:
+        await s3_client.delete_objects(
+            Bucket=bucket,
+            Delete={
+                "Objects": [
+                    {"Key": obj["Key"]} for obj in response["Contents"] if "Key" in obj
+                ]
+            },
+        )
+        response = await s3_client.list_objects_v2(Bucket=bucket)
+
+
+@pytest.fixture
+async def bucket(aiobotocore_s3_client: AioBaseClient, faker: Faker) -> str:
+    response = await aiobotocore_s3_client.create_bucket(Bucket=faker.pystr())
+    assert "ResponseMetadata" in response
+    assert "HTTPStatusCode" in response["ResponseMetadata"]
+    assert response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+    response = await aiobotocore_s3_client.list_buckets()
+    assert response["Buckets"]
+    assert len(response["Buckets"]) == 1
+    bucket_name = response["Buckets"][0]["Name"]
+    yield bucket_name
+    await _clean_bucket_content(aiobotocore_s3_client, bucket_name)
+
+
+@pytest.fixture
+def file_id(faker: Faker) -> str:
+    return faker.pystr()
+
+
+@pytest.fixture
+async def create_upload_links(
+    mocked_s3_server: ThreadedMotoServer,
+    aiobotocore_s3_client: AioBaseClient,
+    faker: Faker,
+    bucket: str,
+    file_id: str,
+) -> AsyncIterator[Callable[[Path, int, ByteSize], Awaitable[FileUploadSchema]]]:
+    file_id = "fake2"
+
+    async def _creator(
+        local_file: Path, num_upload_links: int, chunk_size: ByteSize
+    ) -> FileUploadSchema:
+        local_file.stat().st_size / num_upload_links
+        response = await aiobotocore_s3_client.create_multipart_upload(
+            Bucket=bucket, Key=file_id
+        )
+        assert "UploadId" in response
+        upload_id = response["UploadId"]
+
+        upload_links = parse_obj_as(
+            list[AnyUrl],
+            await asyncio.gather(
+                *[
+                    aiobotocore_s3_client.generate_presigned_url(
+                        "upload_part",
+                        Params={
+                            "Bucket": bucket,
+                            "Key": file_id,
+                            "PartNumber": i + 1,
+                            "UploadId": upload_id,
+                        },
+                    )
+                    for i in range(num_upload_links)
+                ],
+            ),
+        )
+
+        return FileUploadSchema(
+            chunk_size=chunk_size,
+            urls=upload_links,
+            links=FileUploadLinks(
+                abort_upload=parse_obj_as(AnyUrl, faker.uri()),
+                complete_upload=parse_obj_as(AnyUrl, faker.uri()),
+            ),
+        )
+
+    yield _creator
+
+
+@pytest.mark.skip(reason="this will allow to reproduce an issue")
```

Comment:
So currently this test will not fail correct? I would have expected an `xfail` here.

Reply:
no it will not run because it uses a large amount of data.